lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,12 +185,321 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
             tree.create("ABC", source_tarball)
         assert exc.value.tarball == Dataset.stem(source_tarball)
 
-    def test_find(self, selinux_enabled, server_config, make_logger, tarball):
+    def test_unpack(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Test to check the unpacking functionality of the FileTree"""
+        source_tarball, source_md5, md5 = tarball
+
+        def mock_run(args, **kwargs):
+            assert kwargs["capture_output"]
+            assert kwargs["text"]
+
+            tarball_name = source_tarball.name.removesuffix(".tar.xz")
+            tar_target = "--file=" + str(
+                server_config.ARCHIVE / "ABC" / source_tarball.name
+            )
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            assert (
+                kwargs["cwd"]
+                == server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+            )
+
+            assert kwargs["stdin"] == subprocess.DEVNULL
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+            )
+
+        with monkeypatch.context() as m:
+            m.setattr(subprocess, "run", mock_run)
+            m.setattr(shutil, "move", lambda src, dest: dest)
+            m.setattr(shutil, "rmtree", lambda path, **kwargs: None)
```

Comment:
We're missing the mock for `Path.symlink_to()`, here.  😢

(This is the only "success" test case, right?  So, we don't need it in any of the other scenarios...right?)

Reply:
yes